jobserver/forms.py

```diff
@@ -136,7 +141,10 @@ def clean(self):
         if not (repo_url and branch):
             return
 
-        repo = first(self.repos_with_branches, key=lambda r: r["url"] == repo_url)
+        repo = next(
+            (r for r in self.repos_with_branches if r["url"] == repo_url),
+            None,
+        )
```

Comment:
thought: This feels OK to leave as is, with the default set. We explicitly check for that default in the next line.